src/configuration.ts

```diff
@@ -393,6 +393,7 @@ export function setConfigurationCachePath(path: string): void { configurationCac
 // and when the settings watcher detects a change of any properties that may impact the dryrun output.
 export async function readConfigurationCachePath(): Promise<void> {
     // how to get default from package.json to avoid problem with 'undefined' type?
+    let oldConfigurationCachePath = configurationCachePath;
```

Comment:
We need to do this in more places but so far I only focused on what was most annoying in the tests baseline-run diffs. Also, a rewrite/refactoring of the settings will fix many of these problems about the values, non expanded in settings, expanded in memory, out of date versus modified, type validation ...etc... CMake Tools has a great model for settings classes in typescript that we can follow.